gitfile/gitfile.go

```diff
@@ -26,7 +26,7 @@ type GitFile struct {
 }
 
 // initialize with default fetcher
-var gitFile = New(&EnvVarTokenFetcher{})
+var gitFile = New(NewSpiTokenFetcher())
```

Comment:
Why not just get rid of this global variable (and the static function using it) and instead require the caller to hold an instance of the `GitFile` struct? We could provide a constructor like `func Default() *GitFile` or something that would return what we think should be the default configuration of the library.

Reply:
the proposal is to have:

New(TokenFetcher)
and 
Default() (which will do New(&SpiTokenFetcher) under )

and remove the default call `GetFileContents` right and global variable ?